app.rb

```diff
@@ -0,0 +1,84 @@
+require 'sinatra/base'
+require 'sinatra/reloader'
+require './lib/user'
+require './lib/post'
+require './lib/comment'
+require 'sinatra/flash'
+
+class Chitter < Sinatra::Base
+
+  enable :sessions
+  register Sinatra::Flash
+
+  get '/' do
+    @user = User.find_by_id(session[:user_id])
+    @post = Post.order(created_at: :desc).pluck(:peep, :created_at, :id, :user_id)
+    @comment = Comment.order(created_at: :asc)
```

Comment:
This one is a bit more advanced — you can optimise the query to the DB by "telling" AR you're going to need also the comments associated with each post. This way AR will perform a SQL JOIN to retrieve the comments information:
```ruby
@posts = Post.order(created_at: :desc).pluck(:peep, :created_at, :id, :user_id).includes(:comments)
```

And later, when you need to access and list the comments for a given post:
```ruby
post.comments.each do |comment|
  # ...
end
```

This has two advantages:
 * you don't need to filter manually the comments by their post ID
 * the comments are already loaded - otherwise AR would need to make an additional SQL query to SELECT the comments associated to each post, for every post we iterate on in the view. This is a very common pitfall of ActiveRecord (and ORMs in general) [called the N+1 problem](https://blog.appsignal.com/2018/04/24/active-record-performance-the-n-1-queries-antipattern.html), so it's good to learn about it early and recognise when it can happen and how it can be avoided

Reply:
That's really interesting and amazing to know. Thank you! I have just read the article and it makes a lot of sense. Definitely one to keep an eye out for in the future.